src/blob/core.rs

```diff
@@ -517,6 +536,27 @@ impl RawRecords {
     }
 }
 
+pub(crate) fn filter_deleted_headers(headers: Vec<RecordHeader>) -> Vec<RecordHeader> {
+    let deleted = headers.iter().fold(BTreeMap::new(), |mut map, h| {
+        if h.is_deleted() {
+            let entry = map.entry(h.key().to_vec()).or_insert(h.created());
+            *entry = h.created().max(*entry);
+        }
+        map
+    });
+    if deleted.is_empty() {
+        return headers;
+    }
+
+    let mut new_headers = vec![];
+    for header in headers {
+        if deleted.get(header.key()).cloned().unwrap_or_default() < header.created() {
```

Comment:
What happens when the key is not present in `BTreeMap`? 
In this case, the header should normally be copied into `new_headers`.

Reply:
In this case condition will look like `if 0 < header.created()` and this is always true for valid timestamps.